### PR TITLE
ASoC: Intel: acpi: add quirk for APL RVP boards

### PR DESCRIPTION
**Pull request by plbossart**
For some reason the RVP/LeafHill SSDT exposes an INT34C3 ID which is
used on other boards to point to the TDF8532 amplifier. Yay BIOS.

Add a DMI-quirk to ignore this ID and check for other valid machine
driver descriptors.

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>

### CODE REVIEW

File: sound/soc/intel/common/soc-acpi-intel-bxt-match.c [hunk 1]
```diff
@@ -50,6 +84,7 @@ struct snd_soc_acpi_mach snd_soc_acpi_intel_bxt_machines[] = {
 	{
 		.id = "INT34C3",
 		.drv_name = "bxt_tdf8532",
+		.machine_quirk = apl_quirk,
 		.sof_fw_filename = "intel/sof-apl.ri",
 		.sof_tplg_filename = "intel/sof-apl-tdf8532.tplg",
 		.asoc_plat_name = "0000:00:0e.0",
```
Comment (keyonjie) [inline sound/soc/intel/common/soc-acpi-intel-bxt-match.c:90]: so we are planning to add another "INT34C3" entry with different machine_quirk for LeafHill?

Comment (plbossart) [reply sound/soc/intel/common/soc-acpi-intel-bxt-match.c:90]: no, there is a single entry and the quirk determines if this machine descriptor is used as is or replaced by another entry from another table. In the case of LeafHill this codec is not present so it's just discarded.

File: sound/soc/intel/common/soc-acpi-intel-bxt-match.c [hunk 2]
```diff
@@ -6,9 +6,43 @@
...
+};
+
+static struct snd_soc_acpi_mach *apl_quirk(void *arg)
+{
+	struct snd_soc_acpi_mach *mach = arg;
+
+	dmi_check_system(apl_table);
+
+	if (apl_machine_id == APL_TDF8532)
+		return NULL;
```
Comment (keyonjie) [inline sound/soc/intel/common/soc-acpi-intel-bxt-match.c:41]: so this means we will always return NULL and we can never match with INT34C3?

File: sound/soc/intel/common/soc-acpi-intel-bxt-match.c [hunk 3]
```diff
@@ -6,9 +6,43 @@
...
 #include <sound/soc-acpi.h>
 #include <sound/soc-acpi-intel-match.h>
 
+static unsigned long apl_machine_id;
+
+#define APL_TDF8532  1
+
+static int apl_tdf8532_quirk_cb(const struct dmi_system_id *id)
+{
+	apl_machine_id = APL_TDF8532;
```
Comment (keyonjie) [inline sound/soc/intel/common/soc-acpi-intel-bxt-match.c]: apl_machine_id = APL_RVP;
?

Comment (plbossart) [reply sound/soc/intel/common/soc-acpi-intel-bxt-match.c]: yes, makes sense to change. More logical, thanks for the suggestion.

File: sound/soc/intel/common/soc-acpi-intel-bxt-match.c (hunk 2 above)
Comment (plbossart) [reply sound/soc/intel/common/soc-acpi-intel-bxt-match.c:41]: on RVP this ID will not result in any matching for this INT34C3 so the code will keep looking for other IDs and eventually fall to nocodec or HDaudio.
No change on GP-MRB.

**Comment by plbossart** [comment]
updated code with s/tdf8532/rvp as suggested by Keyon.

**Comment by plbossart** [comment]
closing, this will be readded with the backports since the patch is already upstream

**Pull request closed without merging**